### PR TITLE
[FRONT-25] Move frontend deployment to Kubernetes cluster

### LINKED ISSUE #25: Move frontend to DO Kubernetes cluster

### PR DESCRIPTION
**Pull request by cnk1dev**
### Description of issue
Move frontend deployment from Vercel to DigitalOcean Kubernetes cluster

### How I implemented
- Added Dockerfile and .dockerignore to ui package
- Updated docker-compose.yml

### CODE REVIEW

File: packages/ui/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ENV NODE_ENV build
+
+## Setup workspaces
+WORKDIR /app
+COPY package*.json ./
+
+## Install dependencies
+COPY ./packages/ui/package*.json ./packages/ui/
+COPY ./packages/config ./packages/config
+RUN apk --no-cache --virtual build-dependencies add python3 make g++
```
Comment (pczern) [inline packages/ui/Dockerfile:16]: did you need python?

File: packages/ui/Dockerfile
Comment (pczern) [inline packages/ui/Dockerfile:1]: where is npm ci?

File: packages/ui/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,62 @@
+FROM node:18-alpine AS base
+
+# Install dependencies only when needed
+FROM base AS deps
```
Comment (pczern) [inline packages/ui/Dockerfile:4]: is deps needed?

File: packages/ui/Dockerfile
Comment (cnk1dev) [reply packages/ui/Dockerfile:1]: here:
Screenshot 2023-10-22 at 15 20 32

File: packages/ui/Dockerfile (hunk 1 above)
Comment (cnk1dev) [reply packages/ui/Dockerfile:16]: yes, didn't work without it

File: packages/ui/Dockerfile (hunk 2 above)
Comment (cnk1dev) [reply packages/ui/Dockerfile:4]: not sure, I'm not an expert in Dockerfiles but this was also used in the official next.js example

File: packages/ui/Dockerfile (hunk 1 above)
Comment (pczern) [reply packages/ui/Dockerfile:16]: was python in official?

Comment (cnk1dev) [reply packages/ui/Dockerfile:16]: no just `libc6-compat` but that didn't work and I got an error saying python was not found

**Pull request closed without merging**